examples/policies/maintenance-mode/1.0.0/maintenance_mode.lua

```diff
@@ -0,0 +1,37 @@
+-- This is a simple policy. It allows you reject incoming requests with a
+-- specified status code and message.
+-- It's useful for maintance periods or to temporarily block an API
+
+local _M = require('apicast.policy').new('Maintenance mode', '1.0.0')
+
+local tonumber = tonumber
+local new = _M.new
+
+
+function _M.new(configuration)
+  local policy = new(configuration)
+
+  -- Set some default values
+  policy.status_code = 503
+  policy.message = "503 Service Unavailable - Maintenance"
+
+  if configuration then
+    policy.status_code = tonumber(configuration.status) or policy.status_code
+    policy.message = configuration.message or policy.message
+  end
+
+  return policy
+end
+
+function _M:access()
```

Comment:
Both tests fail because this is supposed to be `rewrite`.

APIcast policy rejects the requests in rewrite phase because they don't have authentication. 
So you either want to first require authentication and then show a maintenance page or show maintenance regardless of passed authentication parameters.